src/connection/host_lookup.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * host_lookup.c
+ */
+
+#ident "$Id$"
+
+#include <stdio.h>
+#include <string.h>
+#include <netdb.h>
+#include <limits.h>
+#include <arpa/inet.h>
+
+#include "system_parameter.h"
+#include "environment_variable.h"
+
+#define HOSTNAME_BUF_SIZE              (256)
+
+struct cub_hostent
+{
+  char addr[17];
+  char hostname[HOSTNAME_BUF_SIZE + 1];
+};
+/*
+typedef enum
+{
+  GETHOSTBYNAME = 1,
+  GETHOSTBYNAME_R,
+  GETADDRINFO
+} LOOKUP_FUNC_NAME;
+*/
+
+typedef struct cub_hostent CUB_HOSTENT;
+static CUB_HOSTENT hostent_List[256];	/* "etc/hosts" file maximum line */
+//should I need to allocate all 256 element??
+static const char user_defined_hostfile_Name[] = "hosts.conf";
+
+static int host_conf_element_Count = 0;
+static int host_conf_Use = -1;
+
+static int host_conf_load ();
+static hostent *hostent_compose (char *hostname);
+
+static hostent *
+hostent_compose (char *hostname)
+{
+  static hostent hp;
+  int hostent_list_index, find_index;
+  char addr_transform_buf[17];
+  hp.h_addrtype = AF_INET;
+  hp.h_length = 4;
+  hp.h_name = (char *) malloc (sizeof (char *));
+  hp.h_aliases = (char **) malloc (sizeof (char **));
+  hp.h_aliases[0] = (char *) malloc (sizeof (char *));
+  hp.h_addr_list = (char **) malloc (sizeof (char **));
+  hp.h_addr_list[0] = (char *) malloc (sizeof (char *));
```

Comment:
What about do this after checks the return value of init_pton () calls.

Reply:
I can guess your suggestion means remove unnecessary memory allocation, doesn't it?

If then, I agree and I will fix the code below.
```suggestion
  /**↑↑↑↑↑hostent_List search done↑↑↑↑↑**/
  hp.h_addr_list = (char **) malloc (sizeof (char **));
  hp.h_addr_list[0] = (char *) malloc (sizeof (char *));
  
  if (inet_pton (AF_INET, addr_transform_buf, hp.h_addr_list[0]) != 1){
  free(hp.h_addr_list);   
  free(hp.h_addr_list[0]);
  return err;
  }
 
  hp.h_aliases = (char **) malloc (sizeof (char **));
  hp.h_aliases[0] = (char *) malloc (sizeof (char *));
  
```

or 

```suggestion
   /**↑↑↑↑↑hostent_List search done↑↑↑↑↑**/
  
  if (inet_pton (AF_INET, hostent_List[find_index].addr, temp_buf) != 1){
  return err;
  }
  
  hp.h_aliases = (char **) malloc (sizeof (char **));
  hp.h_aliases[0] = (char *) malloc (sizeof (char *));
  hp.h_addr_list = (char **) malloc (sizeof (char **));
  hp.h_addr_list[0] = (char *) malloc (sizeof (char *));
```

Reply:
I decide to choose second entry. If there are any questions or feedback, please write down the reply :)